libstuff/SSSLState.cpp

```diff
@@ -19,7 +19,9 @@ SSSLState::SSSLState(const string& hostname, int socket) {
     // We need to split it into its componenets.
     string domain;
     uint16_t port;
-    SParseHost(hostname, domain, port);
+    if (!SParseHost(hostname, domain, port)) {
+        STHROW("Invalid host: " + hostname);
+    }
```

Comment:
This caused me a debugging issue because if you pass this a name with no port (i.e., `example.com`) it just fails and sets nothing. This was causing SNI to fail. Throwing the error here will make this sort of thing easier to catch.